### PR TITLE
Feature/remove objc

### PR DESCRIPTION
**Pull request by nathancollard**
The goal of this PR is to update/remove/replace as much objc as possible, but also fixed a few bugs

- Replaced Reachability with Apple's native NWPathMonitor (released in iOS 12)
- Fixed an issue where imageView's weren't showing up in listViews
Screenshot 2023-05-22 at 9 11 03 am

- Converted imageUtil transparent_image to swift equivalent
with_transparency

Example where we don't set uiimageViews highlightedImage property
without_transparency

Note: creating a transparent version within cellForRowAt is computationally heavy, I would recommend either removing it, or if its important look at improving how those images are created (I didn't want to change too much with this PR, just replacing objc features with their swift equivalent)

- Replaced constants with extension's so that xcode can help us where possible.

# Part 2

- Added AutoLayout to several iPad views

example of rotation issues on iPad
before

After AutoLayout change

https://github.com/ODNZSL/nzsl-dictionary-ios/assets/82784626/cfdfd0b9-700f-4e2f-9763-aa81fff7340f

### CODE REVIEW

**Review by joshmcarthur — APPROVED** [review]
Nice, this is a really good improvement, thanks!

File: NZSLDict/Classes/DetailViewController.swift [hunk 1]
```diff
@@ -34,53 +34,83 @@ class DetailViewController: UIViewController, UISplitViewControllerDelegate, UIN
...
-        self.view = view
+        let pbCenterX = playButton.centerXAnchor.constraint(equalTo: videoView.centerXAnchor)
+        let pbCenterY = playButton.centerYAnchor.constraint(equalTo: videoView.centerYAnchor)
+
+        NSLayoutConstraint.activate([
+            pbCenterX, pbCenterY
+        ])
+
+
+//        view.insertSubview(videoView, belowSubview: diagramView)
```
Comment (joshmcarthur) [inline NZSLDict/Classes/DetailViewController.swift:111]: Can we remove the commented out line if it's no longer necessary?

**Review by joshmcarthur — APPROVED** [review]
Nice - tiny little comment, but this change looks great. I'm pretty sure auto layout will help out with a bunch of user feedback

File: NZSLDict/Classes/DetailViewController.swift (hunk 1 above)
Comment (nathancollard) [reply NZSLDict/Classes/DetailViewController.swift:111]: Ah, yes I forgot about that, I left that in as I was going to ask about it, I'm not quite sure why thats in there, I was just going to remove but I wanted to check if there was a reason for it.

**Pull request merged**